openqabot/incrementapprover.py

```diff
@@ -125,41 +136,41 @@ def _check_openqa_jobs(
     def _evaluate_openqa_job_results(
         self,
         results: Dict[str, Dict[str, Dict[str, Any]]],
+        ok_jobs: Set[int],
         not_ok_jobs: Dict[str, Set[str]],
-    ) -> int:
-        ok_jobs = 0
+    ) -> None:
         for state in final_states:
             for result, info in results.get(state, {}).items():
                 if result in ok_results:
-                    ok_jobs += 1
+                    ok_jobs.update(set(info["job_ids"]))
                 else:
                     not_ok_jobs[result].update(info["job_ids"])
-        return ok_jobs
 
     def _evaluate_list_of_openqa_job_results(
         self, list_of_results: List[Dict[str, Dict[str, Dict[str, Any]]]]
-    ) -> Tuple[int, List[str]]:
-        ok_jobs = 0  # count ok jobs
+    ) -> Tuple[Set[int], List[str]]:
+        ok_jobs = set()  # keep track of ok jobs
```

Comment:
This function seems can use the new `ApprovalStatus` class. Both `ok_jobs` and `reasons_to_disapprove` are members of that new class. If I look at the names of the objects I would also say that the `ApprovalStatus` should be a dependency in `IncrementApprover`. 

All this implies a bigger refactoring rather a single change.

Reply:
I don't really know what you're saying. What do you mean exactly when you say class `ApprovalStatus` should be a *dependency* in `IncrementApprover`?